lib/execution_engine2/SDKMethodRunner.py

```diff
@@ -657,7 +710,8 @@ def get_job_status(self, job_id):
             raise ValueError("Please provide valid job_id")
 
         job = self.get_mongo_util().get_job(job_id=job_id)
-        self._test_job_permissions(job, job_id, JobPermissions.READ)
+        if not self._get_job_permission_from_cache(job_id, self.user_id, JobPermissions.READ):
```

Comment:
What do you think about moving `._get_job_permission_from_cache` into  `_test_job_permissions`

Reply:
the problem moving it to `_test_job_permissions` is we cannot limit calls of `self.get_mongo_util().get_job` which is the bottleneck that slows things down. If we move `get_job` into `_test_job_permissions` as well, there are other functions like `cancel_job` does need job after `_test_job_permissions`

Reply:
Ok that makes sense.
What do you think about about moving these 3 lines into a new function called get_job_and_permission or something like that? It can return the job object if necessary, but it hides that you have to make the calls to the cache or not.